docs/interfaces.md

```diff
@@ -0,0 +1,101 @@
+# Interfaces
+
+Interfaces allow snaps to communicate or share resources according to the
+protocol defined by the interface.
+
+Each connection has two ends, a “plug” (consumer) and a “slot” (provider).  A
+plug and a slot can be connected if they use the same interface name.  The
+connection grants necessary permissions for snaps to operate according to the
+protocol.
+
+Slots may support multiple connections to plugs.  For example the OS snap
+exposes the ``network`` slot and all applications that can talk over the
+network connect their plugs there.
+
+Plugs may support multiple connections to slots. There are no examples of this
```

Comment:
Let's drop this until we can explain what it would mean, and make sure the code would work well with it.